src/Umbraco.Core/Models/UserExtensions.cs

```diff
@@ -106,13 +106,14 @@ internal static string[] GetUserAvatarUrls(this IUser user, IAppCache cache)
 
             //use the custom avatar
             var avatarUrl = Current.MediaFileSystem.GetUrl(user.Avatar);
+            var urlGenerator = Current.ImageUrlGenerator;
             return new[]
             {
-                avatarUrl  + "?width=30&height=30&mode=crop",
-                avatarUrl  + "?width=60&height=60&mode=crop",
-                avatarUrl  + "?width=90&height=90&mode=crop",
-                avatarUrl  + "?width=150&height=150&mode=crop",
-                avatarUrl  + "?width=300&height=300&mode=crop"
+                urlGenerator.GetImageUrl(new ImageUrlGenerationOptions(avatarUrl) { ImageCropMode = "crop", Width = 30, Height = 30 }),
```

Comment:
I think it would be useful to have some static factory methods on `ImageUrlGenerationOptions` to avoid this type of duplication for common tasks like cropping. For example: `ImageUrlGenerationOptions.CropUrl(avatarUrl, 30, 30)` would be much simpler.

Reply:
We could even go so far as to make a fluent pattern for this, for example:

`ImageUrlGenerationOptions.Create(avatarUrl).Width(30).Height(30).Crop().Upscale()`

Reply:
I like the fluent idea.  What is the final Upscale call for?  Are you thinking the Create method would return a builder object, and the Upscale call would return the Options object?

If you want to leave the branch open, I can go ahead and implement this today.

Reply:
It's better to submit a new PR on a new branch, makes it easier to review. This was just an example, but `Upscale` would just be setting `upscale=true`. All of this could mostly be done with extension methods since they are just writing to properties so Create could really just return an instance of ImageUrlGenerationOptions. I'm just proposing the idea, not the implementation though, will leave that up to you :)